src/librbd/mirror/GroupGetInfoRequest.cc

```diff
@@ -102,11 +112,17 @@ void GroupGetInfoRequest<I>::handle_get_info(int r) {
     return;
   }
   if (r < 0) {
-    lderr(cct) << "failed to get mirror info of group '" << m_group_name
+    lderr(cct) << "failed to get mirror info of group '" << m_group_id
                << "': " << cpp_strerror(r) << dendl;
     finish(r);
     return;
   }
+  if (m_mirror_group->state == cls::rbd::MIRROR_GROUP_STATE_CREATING) {
+    // No snapshots will have been created and it is likely that the
+    // group has not been created either.
+    finish(0);
```

Comment:
Definitely doesn't sound like we should finish with '0' here, may be finish(-EINVAL); or something making sense.

Reply:
Having no snapshots is a valid state. This is possible before the first non-primary snap is created.